### PR TITLE
Add support for modules that don't support PureConfig's main Scala version

### PR DESCRIPTION
**Pull request by ruippeixotog**
A hacky workaround for #1082. It works as explained in the source code and it will allow us to have Scala 3 only projects (or projects that don't support Scala 2.12 anymore).

I tested this using a test project available in https://github.com/pureconfig/pureconfig/commit/3b5b710e6c8afb4f9b302f26724fff97fe23e771. I checked that both `shapeless3/compile` and `+shapeless3/compile` worked and double-checked using `console`:

```
(sbt) pureconfig ❯ shapeless3/console
[info] Starting scala interpreter...
Welcome to Scala 2.12.15 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_202).
Type in expressions for evaluation. Or try :help.

scala> import pureconfig.shapeless3.Test._
<console>:11: error: object shapeless3 is not a member of package pureconfig
       import pureconfig.shapeless3.Test._
                         ^
[success] Total time: 19 s, completed Oct 8, 2021 7:47:57 PM

(sbt) pureconfig ❯ +shapeless3/console
[info] Setting Scala version to 3.0.2 on 5 projects.
[info] Excluded 47 projects, run ++ 3.0.2 -v for more details.
[info] Reapplying settings...
[info] set current project to pureconfig (in build file:/Users/rui/Documents/Dev/own/pureconfig/)
[info] Formatting 7 Scala sources...
Default dialect is deprecated; use explicit: [Scala211,scala212,Scala212Source3,scala213,Scala213Source3,Sbt0137,Sbt1,scala3]
Also see https://scalameta.org/scalafmt/docs/configuration.html#scala-dialects"

scala> import pureconfig.shapeless3.Test._

scala> "hello".appendRui
val res0: String = hello Rui
```

I'm sure there we'll find more edge cases as we develop projects using this. For now, this seems like a good start.

### CODE REVIEW

File: project/MainScalaVersionHack.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  * In order to prevent projects like these from breaking builds, the settings defined here effectively generate a dummy
+  * project with no source files nor library dependencies. This will ensure SBT doesn't try to fetch non-existent
+  * dependencies and compile module's code on an unsupported Scala version.
+  *
+  * `projectSettings` should be included as the last statement in a module's `build.sbt`. Modules using these settings
+  * do not need to define `scalaVersion`. `crossScalaVersions` should include only the Scala versions actually supported
+  * by the project.
+  */
+object MainScalaVersionHack {
+  val isDummyProject = settingKey[Boolean]("Whether this is a dummy project generated by MainScalaVersionFix.scala")
```
Comment (jcazevedo) [inline project/MainScalaVersionHack.scala]: To match the filename.

```suggestion
  val isDummyProject = settingKey[Boolean]("Whether this is a dummy project generated by MainScalaVersionHack.scala")
```

**Review by jcazevedo — COMMENTED** [review]
Looks good. Thanks for this! Just commented on a minor issue in the `isDummyProject` description.

File: project/MainScalaVersionHack.scala (hunk 1 above)
Comment (ruippeixotog) [reply project/MainScalaVersionHack.scala]: Fixed!

**Review by jcazevedo — APPROVED** [review]
Thanks for this!
